bindata/oauth-openshift/deployment.yaml

```diff
@@ -0,0 +1,141 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  namespace: openshift-authentication
+  name: oauth-openshift
+  labels:
+    app: oauth-openshift
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: oauth-openshift
+  template:
+    metadata:
+      namespace: openshift-authentication
+      name: oauth-openshift
+      labels:
+        app: oauth-openshift
+    spec:
+      serviceAccountName: oauth-openshift
+      nodeSelector:
+        node-role.kubernetes.io/master: ''
+      priorityClassName: system-cluster-critical
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchLabels:
+                    app: oauth-openshift
+                topologyKey: kubernetes.io/hostname
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: NoSchedule
+        - key: node.kubernetes.io/unreachable
+          operator: Exists
+          effect: NoExecute
+          tolerationSeconds: 120
+        - key: node.kubernetes.io/not-ready
+          operator: Exists
+          effect: NoExecute
+          tolerationSeconds: 120
+      containers:
+        - name: oauth-openshift
+          image: ${IMAGE}
+          command:
+            - /bin/bash
+            - '-ec'
+          args:
+            - >
+              if [ -s
+              /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt
+              ]; then
+                  echo "Copying system trust bundle"
+                  cp -f /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
+              fi
+              exec oauth-server osinserver
+              --config=/var/config/system/configmaps/v4-0-config-system-cliconfig/v4-0-config-system-cliconfig
+              --v=${LOG_LEVEL}
```

Comment:
These line get squashed in the real deployment so that the args look like this:
```
      if [ -s /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt ]; then
          echo "Copying system trust bundle"
          cp -f /var/config/system/configmaps/v4-0-config-system-trusted-ca-bundle/ca-bundle.crt /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
      fi exec oauth-server osinserver --config=/var/config/system/configmaps/v4-0-config-system-cliconfig/v4-0-config-system-cliconfig --v=2
```
As @soltysh pointed out earlier in his yaml-multiline findings, you may want to start with
```
args:
  - |
```
which is supposed to keep the newlines.

See https://yaml-multiline.info/ which is quite helpful

You'll have to add backslashes before each line ending in the `exec` part I think

Reply:
@stlaz what do you think of running the "copy system bundle" part in an init container? Then the cmd/args for this container will be more maintainable?

Reply:
Wouldn't you need a volume for such a change to persist?

Reply:
Yes, I think an EmptyDir volume would do as this is the only file expected in the target directory. But I'll leave that for another day since this is working as is.